python/analysis/raster/qgsalignraster.sip

```diff
@@ -1,183 +1,325 @@
+ ************************************************************************/
+
+
+
+
+typedef void *GDALDatasetH;
+
 
 class QgsAlignRaster
 {
-%TypeHeaderCode
-#include <qgsalignraster.h>
-#include <gdal_version.h>
+%Docstring
+ QgsAlignRaster takes one or more raster layers and warps (resamples) them
+ so they have the same:
+ - coordinate reference system
+ - cell size and raster size
+ - offset of the raster grid
+
+.. versionadded:: 2.12
 %End
 
+%TypeHeaderCode
+#include "qgsalignraster.h"
+%End
   public:
     QgsAlignRaster();
 
-    //! Utility class for gathering information about rasters
     struct RasterInfo
     {
-    public:
-      //! Construct raster info with a path to a raster file
-      RasterInfo( const QString &layerpath );
-      ~RasterInfo();
-
-      //! Check whether the given path is a valid raster
-      bool isValid() const;
-
-      //! Return CRS in WKT format
-      QString crs() const;
-      //! Return size of the raster grid in pixels
-      QSize rasterSize() const;
-      //! Return number of raster bands in the file
-      int bandCount() const;
-      //! Return cell size in map units
-      QSizeF cellSize() const;
-      //! Return grid offset
-      QPointF gridOffset() const;
-      //! Return extent of the raster
-      QgsRectangle extent() const;
-      //! Return origin of the raster
-      QPointF origin() const;
-
-      //! write contents of the object to standard error stream - for debugging
-      void dump() const;
-
-      //! Get raster value at the given coordinates (from the first band)
-      double identify( double mx, double my );
-
-      private:
-        RasterInfo( const QgsAlignRaster::RasterInfo &rh );
+      public:
+        RasterInfo( const QString &layerpath );
+%Docstring
+Construct raster info with a path to a raster file
+%End
+        ~RasterInfo();
+
+
+        bool isValid() const;
+%Docstring
+Check whether the given path is a valid raster
+ :rtype: bool
+%End
+
+        QString crs() const;
+%Docstring
+Return CRS in WKT format
+ :rtype: str
+%End
+        QSize rasterSize() const;
+%Docstring
+Return size of the raster grid in pixels
+ :rtype: QSize
+%End
+        int bandCount() const;
+%Docstring
+Return number of raster bands in the file
+ :rtype: int
+%End
+        QSizeF cellSize() const;
+%Docstring
+Return cell size in map units
+ :rtype: QSizeF
+%End
+        QPointF gridOffset() const;
+%Docstring
+Return grid offset
+ :rtype: QPointF
+%End
+        QgsRectangle extent() const;
+%Docstring
+Return extent of the raster
+ :rtype: QgsRectangle
+%End
+        QPointF origin() const;
+%Docstring
+Return origin of the raster
+ :rtype: QPointF
+%End
+
+        void dump() const;
+%Docstring
+write contents of the object to standard error stream - for debugging
+%End
+
+        double identify( double mx, double my );
+%Docstring
+Get raster value at the given coordinates (from the first band)
+ :rtype: float
+%End
+
+      protected:
+        int mXSize, mYSize;
```

Comment:
@3nids looks like there's an issue here in sipify - it's included these protected member variables